cms/templates/widgets/deprecated-course-key-warning.html

```diff
@@ -0,0 +1,16 @@
+<%!
+from django.conf import settings
+from django.utils.translation import ugettext as _
+is_enabled = settings.FEATURES.get('SHOW_STUDIO_WARNING_FOR_DEPRECATED_COURSE_KEYS', False)
+%>
+
+%if is_enabled and context_course and (context_course.id.deprecated or True):
+  <div class="wrapper wrapper-alert wrapper-alert-error is-shown">
+    <div class="alert announcement">
+      <span class="feedback-symbol fa fa-warning" aria-hidden="true"></span><span class="sr">${_("Warning")}</span>
+      <div class="copy">
+        <h2 class="title title-3 warning-heading-text">${_("This course uses deprecated course keys; that's bad!")}</h2>
```

Comment:
TODO: Add better text!

Reply:
This course uses a legacy storage format and will be removed from {platform/site-name} on September 1st, 2020. Please reach out to your support team contact if you have any additional questions or concerns. <-- this date might need to be configurable?  

I'm ok not linking to specific documentation, message is general enough to hopefully work for open edx as well.

Reply:
@marcotuts I tweaked message to split the first part out of the body and into the header. Let me know.

This feature now supports 3 configuration options:
- `False` to disable (no screenshot needed)
- `True` to enable with an open-ended deprecation date [1]
- `'2020-09-01'` (an ISO-8601/`yyyy-mm-dd` string) to enable with an explicit deprecation date [2]

I don't think we can list an explicit date that would apply for all (or _any_, really) community instances. 
- Most/many instances won't be running Juniper live in production before the expiration anyway (here, September).
- We don't advertise release dates for named releases, so we can't predict when community members would upgrade to Koa, even if they did immediately upon its release, so this message will be outdated for most users, most of the time.

As drawn up now, the default configuration for the community (again, edx.org will disable it to start) will be to enable the warning _but_ with an open-ended support deadline. If operators want, they can update this setting to an explicit deadline if/when they have one.

For us, we'll disable the feature entirely at launch. Similarly, we can chose to enable it with an explicit support-drop date, if/when needed.

@marcotuts Let me know what you think of that ^

- [1] Open-ended support date
    ![Screenshot_2020-06-16_14-20-35](https://user-images.githubusercontent.com/7300975/84831660-9a086d00-afe0-11ea-9c49-d075f195a1ec.png)
- [2] Explicit support date
    ![Screenshot_2020-06-16_14-21-16](https://user-images.githubusercontent.com/7300975/84831682-a260a800-afe0-11ea-954d-e858fba25a0e.png)

Reply:
This looks great, and I think we for edX.org purposes would benefit from setting an explicit date if that configuration isn't too much of a hassle to support fully.  (2 - edx.org, 1 - Open edx default)

Reply:
Nope, super easy. The feature is already toggled off for all of our instances, so when we decide to go live, just tell me the desired date.
It's as easy as setting the flag to `'2020-09-01'` instead of `True`; the magic is built in :)